# HG changeset patch
# User =?UTF-8?q?Yoshiki=20V=C3=A1zquez=20Baeza?= <yoshiki89@gmail.com>
# Date 1461443630 25200
#      Sat Apr 23 13:33:50 2016 -0700
# Branch gentoo-1.7
# Node ID 3d70abf517990f44165dee996b6bf0e089b43cbc
# Parent  32f2c83c36541cb655f4ef41e5ebeae376b17203
[PATCH] feature: new-mail
From 6eca9b12e4262479f4f2a41581a892d7e5e1329d Mon Sep 17 00:00:00 2001
This setting executes a command when new mail is received.
---
 curs_main.c | 12 ++++++++++++
 globals.h   |  1 +
 init.h      |  7 +++++++
 3 files changed, 20 insertions(+)

diff -r 32f2c83c3654 -r 3d70abf51799 curs_main.c
--- a/curs_main.c	Thu Jun 02 23:26:09 2016 +0100
+++ b/curs_main.c	Sat Apr 23 13:33:50 2016 -0700
@@ -809,6 +809,12 @@ int mutt_index_menu (void)
 	  mutt_message _("New mail in this mailbox.");
 	  if (option (OPTBEEPNEW))
 	    beep ();
+	  if (NewMailCmd)
+	  {
+	    char cmd[LONG_STRING];
+	    menu_status_line(cmd, sizeof(cmd), menu, NONULL(NewMailCmd));
+	    mutt_system(cmd);
+	  }
 	} else if (check == MUTT_FLAGS)
 	  mutt_message _("Mailbox was externally modified.");
 
@@ -835,6 +841,12 @@ int mutt_index_menu (void)
          menu->redraw |= REDRAW_STATUS;
          if (option (OPTBEEPNEW))
            beep();
+         if (NewMailCmd)
+         {
+           char cmd[LONG_STRING];
+           menu_status_line(cmd, sizeof(cmd), menu, NONULL(NewMailCmd));
+           mutt_system(cmd);
+         }
        }
      }
      else
diff -r 32f2c83c3654 -r 3d70abf51799 globals.h
--- a/globals.h	Thu Jun 02 23:26:09 2016 +0100
+++ b/globals.h	Sat Apr 23 13:33:50 2016 -0700
@@ -123,6 +123,7 @@ WHERE char *Postponed;
 WHERE char *PostponeEncryptAs;
 WHERE char *Prefix;
 WHERE char *PrintCmd;
+WHERE char *NewMailCmd;
 WHERE char *QueryCmd;
 WHERE char *QueryFormat;
 WHERE char *Realname;
diff -r 32f2c83c3654 -r 3d70abf51799 init.h
--- a/init.h	Thu Jun 02 23:26:09 2016 +0100
+++ b/init.h	Sat Apr 23 13:33:50 2016 -0700
@@ -307,6 +307,13 @@ struct option_t MuttVars[] = {
   { "beep",		DT_BOOL, R_NONE, OPTBEEP, 1 },
   /*
   ** .pp
+  ** If \fIset\fP, Mutt will call this command after a new message is received.
+  ** See the $$status_format documentation for the values that can be formatted
+  ** into this command.
+  */
+  { "new_mail_command",	DT_PATH, R_NONE, UL &NewMailCmd, UL NULL },
+  /*
+  ** .pp
   ** When this variable is \fIset\fP, mutt will beep when an error occurs.
   */
   { "beep_new",		DT_BOOL, R_NONE, OPTBEEPNEW, 0 },
